test/integration/test/ip_camera_test.js

```diff
@@ -0,0 +1,118 @@
+var Prepper = require('../lib/testPrepper.js');
+
+Prepper.makeSuite('Control lights locally',function(){
+  var header = new Config('./test/integration/lib/config/header.js');
+  var credential = config.get('credential'); 
+
+  var setupURL = config.get('foxbox.url') + '/users/setup';
+  var loginURL = config.get('foxbox.url') + '/users/login';
+  var serviceURL = config.get('foxbox.url') + '/api/v1';
+  var serviceListURL = serviceURL + '/services'; 
+  var getterURL = serviceURL + '/channels/get';
+  var setterURL = serviceURL + '/channels/set';
+  var cameraService;
+
+  it('create a new credential',function(){
+
+    return chakram.post(setupURL,credential)
+    .then(function(setupResp){
+      expect(setupResp).to.have.status(201);
+    });
+  });
+
+  it('login to foxbox',function(){
+    var encoded_cred = new Buffer(credential.username+
+      ':'+credential.password).toString('base64');
+
+    // supply the credential used in previous test
+    header.Authorization = 'Basic ' + encoded_cred;
+
+    return  chakram.post(loginURL,null,{'headers' : header})
+    .then(function(loginResp){
+      expect(loginResp).to.have.status(201);
+    });
+  });
+
+    describe ('once logged in', function () {
+      
+      before(function() {
+        return chakram.post(loginURL,null,{'headers' : header})
+        .then(function(resp){
+         header.Authorization = 'Bearer ' + resp.body.session_token;
+         chakram.setRequestDefaults({'headers': header});
+       });
+      });
+
+      it('check simulated camera is registered',function(){        
+        return chakram.get(serviceListURL)
+        .then(function(listResponse) {
+          for (var entry of listResponse.body) {
+            if (entry.adapter === 'ip-camera@link.mozilla.org') {
+                cameraService = entry;                
+             }
+          }
+          expect(cameraService.properties.udn).
+          equals(config.get('ipCamera.udn'));
+          expect(listResponse).to.have.status(200);
+        });
+      });
+      
+      it('take a picture',function(){
+        var setter = 'setter:' + 
+        cameraService.id.replace('service:','snapshot.');
+        var payload = {'select': {'id': setter}, 'value': {'Unit': {}}};
+        return chakram.put(setterURL, payload)
+        .then(function(cmdResponse) {
+          expect(cmdResponse).to.have.status(200);
+          expect(cmdResponse.body[setter]).equals(null);
+        });
+      }); 
+
+      // Image filesnames have YYYY-MM-DD prefix. make sure the last image
+      // taken is from today
+      it('get a list of images taken',function(){        
+        var date = new Date();
+        var year = date.getFullYear();
+        var month = date.getMonth() + 1;
+        month = (month < 10 ? '0' : '') + month;
+        var day = date.getDate();
+        day = (day < 10 ? '0' : '') + day;
+        var datePrefix = year + '-' +month +'-' + day +'-';
```

Comment:
We might need such logic later. I'd recommend to use a library [like this one](https://www.npmjs.com/package/iso-date). This would save these 6 lines, I think